src/toil/fileStores/__init__.py

```diff
@@ -28,32 +29,35 @@ class FileID(str):
     def __new__(cls, fileStoreID, *args):
         return super(FileID, cls).__new__(cls, fileStoreID)
 
-    def __init__(self, fileStoreID, size):
+    def __init__(self, fileStoreID, size, executable=False):
         # Don't pass an argument to parent class's __init__.
         # In Python 3 we can have super(FileID, self) hand us object's __init__ which chokes on any arguments.
         super(FileID, self).__init__()
         self.size = size
+        self.executable = executable
 
     def pack(self):
         """
         Pack the FileID into a string so it can be passed through external code.
         """
-        return '{}:{}'.format(self.size, self)
+        return '{}:{}:{}'.format(self.size, int(self.executable), self)
 
     @classmethod
     def forPath(cls, fileStoreID, filePath):
-        return cls(fileStoreID, os.stat(filePath).st_size)
+        executable = os.stat(filePath).st_mode & stat.S_IXUSR != 0
+        return cls(fileStoreID, os.stat(filePath).st_size, executable)
 
     @classmethod
     def unpack(cls, packedFileStoreID):
         """
         Unpack the result of pack() into a FileID object.
         """
 
-        # Find the separating character (first instance)
-        sepPos = packedFileStoreID.find(':')
+        # Separate each part of fileID
+        vals = packedFileStoreID.split(':')
```

Comment:
The reason we were using find before is that we can't guarantee that there are no colons in the File ID itself. If you want to use split, you need to make sure to grab the first and last items and paste the rest back together with colons, or else encode/escape internal colons and then decode/unescape them here.

I don't think this will fail in practice, because hardly anybody is going to use colons in filenames anywhere, and all the current file store implementations might avoid them, but it doesn't work in theory.

Reply:
Another option would be to just ban colons in the file ID with an `assert ':' not in self, 'Colons not allowed in ' + self` in `__init__`, and a suitable comment. Then it would be up to the job store implementations to make sure colons are properly escaped/unescaped when making/interpreting FileIDs.